### PR TITLE
RPC method to get block by height

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

MetaMask requests the block (we're not 100% why, but likely to check a tx has been incorporated successfully).

### What changes were made as part of this PR:

Functional.

- Adds RPC method to get block
- Adds enclave method to retrieve rollup by height

### CODE REVIEW

File: go/enclave/enclave.go [hunk 1]
```diff
@@ -319,6 +320,33 @@ func (e *enclaveImpl) GetRollup(rollupHash common.L2RootHash) *common.ExtRollup
 	return nil
 }
 
+func (e *enclaveImpl) GetRollupByHeight(rollupHeight uint64) *common.ExtRollup {
+	rollupHeightBig := big.NewInt(int64(rollupHeight))
+
+	// TODO - Consider improving efficiency by directly fetching rollup by number.
+	rollup := e.storage.FetchHeadRollup()
+	for {
+		if rollup.Number().Cmp(rollupHeightBig) == 0 {
```
Comment (tudor-malene) [inline go/enclave/enclave.go:329]: you can use: `NumberU64` here and make the comparison easier

File: go/host/client_api_eth.go [hunk 2]
```diff
@@ -43,14 +43,22 @@ func (api *EthereumAPI) GetBalance(_ context.Context, encryptedParams common.Enc
...
-func (api *EthereumAPI) GetBlockByNumber(context.Context, rpc.BlockNumber, bool) (map[string]interface{}, error) {
-	result := map[string]interface{}{
-		// TODO - Return non-dummy values.
-		"baseFeePerGas": (*hexutil.Big)(big.NewInt(0)),
-		"number":        (*hexutil.Big)(big.NewInt(0)),
+// GetBlockByNumber returns the rollup with the given height as a block. No transactions are included.
+func (api *EthereumAPI) GetBlockByNumber(_ context.Context, number rpc.BlockNumber, _ bool) (map[string]interface{}, error) {
+	extRollup := api.host.EnclaveClient.GetRollupByHeight(uint64(number))
+
+	block := map[string]interface{}{
```
Comment (tudor-malene) [inline go/host/client_api_eth.go:50]: there's a field missing which we have in the header:
- transactionsroot

**Review by tudor-malene — COMMENTED** [review]
lgtm. minor comments.

File: go/common/enclave.go [hunk 3]
```diff
@@ -68,6 +68,9 @@ type Enclave interface {
 	// GetRollup returns the rollup with the given hash
 	GetRollup(rollupHash L2RootHash) *ExtRollup
 
+	// GetRollupByHeight returns the rollup with the given height.
```
[the remark this answers is not in the document]
Comment (joeldudley) [reply go/common/enclave.go:71]: Fixed.

File: go/host/client_api_eth.go (hunk 2 above)
Comment (joeldudley) [reply go/host/client_api_eth.go:50]: Think I've added it now. Is it `extRollup.Header.TxHash`?

File: go/enclave/enclave.go [hunk 4]
```diff
@@ -326,12 +326,12 @@ func (e *enclaveImpl) GetRollupByHeight(rollupHeight uint64) *common.ExtRollup {
 	// TODO - Consider improving efficiency by directly fetching rollup by number.
 	rollup := e.storage.FetchHeadRollup()
 	for {
-		if rollup.Number().Cmp(rollupHeightBig) == 0 {
+		if rollup.Number().Uint64() == 0 {
```
Comment (tudor-malene) [inline go/enclave/enclave.go:329]: I don't think this is correct.
Shoul be: `if rollup.NumberU64() == rollupHeight`

**Review by tudor-malene — COMMENTED** [review]
one comparison is prob incorrect

**Review by tudor-malene — APPROVED** [review]
lgtm

**Pull request merged**